Can't build extensions on Windows

Probably an issue with npm. May be related to https://github.com/bokeh/bokeh/issues/13834. Fails with:
```py
================================== FAILURES ===================================
______________________________ test_ext_commands ______________________________
tmpdir = local('C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_ext_commands0')
    def test_ext_commands(tmpdir) -> None:
        tmp = str(tmpdir.mkdir("bk_ext_01"))
    
        assert _names(tmp) == []
    
        assert ext.init(tmp, bokehjs_version="3.0.0") is True
        assert _names(tmp) == [
            "bokeh.ext.json",
            "index.ts",
            "package.json",
            "tsconfig.json",
        ]
    
>       assert ext.build(tmp) is True
E       AssertionError: assert False is True
E        +  where False = <function build at 0x0000023045EEC3A0>('C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_ext_commands0\\bk_ext_01')
E        +    where <function build at 0x0000023045EEC3A0> = ext.build
tests\unit\bokeh\test_ext.py:46: AssertionError
---------------------------- Captured stdout call -----------------------------
Working directory: C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_ext_commands0\bk_ext_01
Wrote C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_ext_commands0\bk_ext_01\bokeh.ext.json
Wrote C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_ext_commands0\bk_ext_01\package.json
Wrote C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_ext_commands0\bk_ext_01\tsconfig.json
Created empty index.ts. This is the entry point of your extension.
You can build your extension with bokeh build C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_ext_commands0\bk_ext_01
All done.
Working directory: C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_ext_commands0\bk_ext_01
Using different version of bokeh, rebuilding from scratch.
Running npm install.
TypeScript lib: C:\Miniconda3\envs\bk-test\Lib\site-packages\bokeh\server\static\lib
Using C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_ext_commands0\bk_ext_01\tsconfig.json
Compiling styles (0 files)
Compiling TypeScript (1 file)
Error: base path C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_ext_commands0\bk_ext_01\node_modules doesn't exist or isn't a directory
    at new Linker (C:\Miniconda3\envs\bk-test\Lib\site-packages\bokeh\server\static\js\compiler.js:184514:27)
    at build (C:\Miniconda3\envs\bk-test\Lib\site-packages\bokeh\server\static\js\compiler.js:183168:24)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async main (C:\Miniconda3\envs\bk-test\Lib\site-packages\bokeh\server\static\js\compiler.js:222:32)
__________ Test_bundle_custom_extensions.test_with_INLINE_resources ___________
self = <test_bundle.Test_bundle_custom_extensions object at 0x0000023049624C10>
    def test_with_INLINE_resources(self) -> None:
        from latex_label import LatexLabel
        plot = models.Plot()
        plot.add_layout(LatexLabel(x=0, y=0))
>       bundle = beb.bundle_for_objs_and_resources([plot], INLINE)
tests\unit\bokeh\embed\test_bundle.py:118: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\Miniconda3\envs\bk-test\lib\site-packages\bokeh\embed\bundle.py:195: in bundle_for_objs_and_resources
    js_raw.extend([ Resources._inline(bundle.artifact_path) for bundle in extensions ])
C:\Miniconda3\envs\bk-test\lib\site-packages\bokeh\embed\bundle.py:195: in <listcomp>
    js_raw.extend([ Resources._inline(bundle.artifact_path) for bundle in extensions ])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
path = WindowsPath('D:/a/bokeh/bokeh/tests/unit/bokeh/embed/latex_label/dist/latex_label.js')
    @staticmethod
    def _inline(path: Path) -> str:
        filename = path.name
        begin = f"/* BEGIN {filename} */"
>       with open(path, "rb") as f:
E       FileNotFoundError: [Errno 2] No such file or directory: 'D:\\a\\bokeh\\bokeh\\tests\\unit\\bokeh\\embed\\latex_label\\dist\\latex_label.js'
C:\Miniconda3\envs\bk-test\lib\site-packages\bokeh\resources.py:479: FileNotFoundError
---------------------------- Captured stdout setup ----------------------------
Working directory: D:\a\bokeh\bokeh\tests\unit\bokeh\embed\latex_label
Using different version of bokeh, rebuilding from scratch.
Running npm install.
TypeScript lib: C:\Miniconda3\envs\bk-test\Lib\site-packages\bokeh\server\static\lib
Using D:\a\bokeh\bokeh\tests\unit\bokeh\embed\latex_label\tsconfig.json
Compiling styles (0 files)
Compiling TypeScript (2 files)
tests/unit/bokeh/embed/latex_label/index.ts:3:31 - error TS2307: Cannot find module '@bokehjs/base' or its corresponding type declarations.
3 import {register_models} from "@bokehjs/base"
                                ~~~~~~~~~~~~~~~
tests/unit/bokeh/embed/latex_label/latex_label.ts:1:40 - error TS2307: Cannot find module '@bokehjs/models/annotations/html/label' or its corresponding type declarations.
1 import {HTMLLabel, HTMLLabelView} from "@bokehjs/models/annotations/html/label"
                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tests/unit/bokeh/embed/latex_label/latex_label.ts:3:24 - error TS2307: Cannot find module 'katex' or its corresponding type declarations.
3 import * as katex from "katex"
                         ~~~~~~~
tests/unit/bokeh/embed/latex_label/latex_label.ts:11:22 - error TS[411](https://github.com/bokeh/bokeh/actions/runs/8858943839/job/24343080009#step:5:412)2: This member cannot have an 'override' modifier because its containing class 'LatexLabelView' does not extend another class.
11   protected override _render(): void {
                        ~~~~~~~
tests/unit/bokeh/embed/latex_label/latex_label.ts:13:29 - error TS2339: Property 'text' does not exist on type 'LatexLabel'.
13     katex.render(this.model.text, this.el, {displayMode: true})
                               ~~~~
tests/unit/bokeh/embed/latex_label/latex_label.ts:13:40 - error TS2339: Property 'el' does not exist on type 'LatexLabelView'.
13     katex.render(this.model.text, this.el, {displayMode: true})
                                          ~~
tests/unit/bokeh/embed/latex_label/latex_label.ts:18:19 - error TS4112: This member cannot have an 'override' modifier because its containing class 'LatexLabel' does not extend another class.
18   static override __module__ = "latex_label"
                     ~~~~~~~~~~
tests/unit/bokeh/embed/latex_label/latex_label.ts:21:20 - error TS2339: Property 'default_view' does not exist on type 'LatexLabel'.
21     this.prototype.default_view = LatexLabelView
                      ~~~~~~~~~~~~
Error: base path D:\a\bokeh\bokeh\tests\unit\bokeh\embed\latex_label\node_modules doesn't exist or isn't a directory
    at new Linker (C:\Miniconda3\envs\bk-test\Lib\site-packages\bokeh\server\static\js\compiler.js:184514:27)
    at build (C:\Miniconda3\envs\bk-test\Lib\site-packages\bokeh\server\static\js\compiler.js:183168:24)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async main (C:\Miniconda3\envs\bk-test\Lib\site-packages\bokeh\server\static\js\compiler.js:222:32)
_________ Test_bundle_ext_package_no_main.test_with_INLINE_resources __________
self = <test_bundle.Test_bundle_ext_package_no_main object at 0x0000023049625330>
    def test_with_INLINE_resources(self) -> None:
        from ext_package_no_main import AModel
        model = AModel()
>       bundle = beb.bundle_for_objs_and_resources([model], INLINE)
tests\unit\bokeh\embed\test_bundle.py:165: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\Miniconda3\envs\bk-test\lib\site-packages\bokeh\embed\bundle.py:195: in bundle_for_objs_and_resources
    js_raw.extend([ Resources._inline(bundle.artifact_path) for bundle in extensions ])
C:\Miniconda3\envs\bk-test\lib\site-packages\bokeh\embed\bundle.py:195: in <listcomp>
    js_raw.extend([ Resources._inline(bundle.artifact_path) for bundle in extensions ])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
path = WindowsPath('D:/a/bokeh/bokeh/tests/unit/bokeh/embed/ext_package_no_main/dist/ext_package_no_main.js')
    @staticmethod
    def _inline(path: Path) -> str:
        filename = path.name
        begin = f"/* BEGIN {filename} */"
>       with open(path, "rb") as f:
E       FileNotFoundError: [Errno 2] No such file or directory: 'D:\\a\\bokeh\\bokeh\\tests\\unit\\bokeh\\embed\\ext_package_no_main\\dist\\ext_package_no_main.js'
C:\Miniconda3\envs\bk-test\lib\site-packages\bokeh\resources.py:479: FileNotFoundError
---------------------------- Captured stdout setup ----------------------------
Working directory: D:\a\bokeh\bokeh\tests\unit\bokeh\embed\ext_package_no_main
Using different version of bokeh, rebuilding from scratch.
Running npm install.
TypeScript lib: C:\Miniconda3\envs\bk-test\Lib\site-packages\bokeh\server\static\lib
Using D:\a\bokeh\bokeh\tests\unit\bokeh\embed\ext_package_no_main\tsconfig.json
Compiling styles (0 files)
Compiling TypeScript (1 file)
tests/unit/bokeh/embed/ext_package_no_main/index.ts:1:21 - error TS2307: Cannot find module '@bokehjs/model' or its corresponding type declarations.
1 import {Model} from "@bokehjs/model"
                      ~~~~~~~~~~~~~~~~
tests/unit/bokeh/embed/ext_package_no_main/index.ts:4:19 - error TS4112: This member cannot have an 'override' modifier because its containing class 'AModel' does not extend another class.
4   static override __module__ = "ext_package_no_main"
                    ~~~~~~~~~~
tests/unit/bokeh/embed/ext_package_no_main/index.ts:7:31 - error TS2307: Cannot find module '@bokehjs/base' or its corresponding type declarations.
7 import {register_models} from "@bokehjs/base"
                                ~~~~~~~~~~~~~~~
Error: base path D:\a\bokeh\bokeh\tests\unit\bokeh\embed\ext_package_no_main\node_modules doesn't exist or isn't a directory
    at new Linker (C:\Miniconda3\envs\bk-test\Lib\site-packages\bokeh\server\static\js\compiler.js:18[451](https://github.com/bokeh/bokeh/actions/runs/8858943839/job/24343080009#step:5:452)4:27)
    at build (C:\Miniconda3\envs\bk-test\Lib\site-packages\bokeh\server\static\js\compiler.js:183168:24)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async main (C:\Miniconda3\envs\bk-test\Lib\site-packages\bokeh\server\static\js\compiler.js:222:32)
___________ Test_bundle_ext_package_no_main.test_with_CDN_resources ___________
self = <test_bundle.Test_bundle_ext_package_no_main object at 0x0000023049625480>
    def test_with_CDN_resources(self) -> None:
        from ext_package_no_main import AModel
        model = AModel()
>       bundle = beb.bundle_for_objs_and_resources([model], CDN)
tests\unit\bokeh\embed\test_bundle.py:172: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\Miniconda3\envs\bk-test\lib\site-packages\bokeh\embed\bundle.py:203: in bundle_for_objs_and_resources
    js_raw.append(Resources._inline(bundle.artifact_path))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
path = WindowsPath('D:/a/bokeh/bokeh/tests/unit/bokeh/embed/ext_package_no_main/dist/ext_package_no_main.js')
    @staticmethod
    def _inline(path: Path) -> str:
        filename = path.name
        begin = f"/* BEGIN {filename} */"
>       with open(path, "rb") as f:
E       FileNotFoundError: [Errno 2] No such file or directory: 'D:\\a\\bokeh\\bokeh\\tests\\unit\\bokeh\\embed\\ext_package_no_main\\dist\\ext_package_no_main.js'
C:\Miniconda3\envs\bk-test\lib\site-packages\bokeh\resources.py:479: FileNotFoundError
---------- coverage: platform win32, python 3.10.14-final-0 ----------
Coverage XML written to file coverage.xml
=========================== short test summary info ===========================
FAILED tests/unit/bokeh/test_ext.py::test_ext_commands - AssertionError: assert False is True
 +  where False = <function build at 0x0000023045EEC3A0>('C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_ext_commands0\\bk_ext_01')
 +    where <function build at 0x0000023045EEC3A0> = ext.build
FAILED tests/unit/bokeh/embed/test_bundle.py::Test_bundle_custom_extensions::test_with_INLINE_resources - FileNotFoundError: [Errno 2] No such file or directory: 'D:\\a\\bokeh\\bokeh\\tests\\unit\\bokeh\\embed\\latex_label\\dist\\latex_label.js'
FAILED tests/unit/bokeh/embed/test_bundle.py::Test_bundle_ext_package_no_main::test_with_INLINE_resources - FileNotFoundError: [Errno 2] No such file or directory: 'D:\\a\\bokeh\\bokeh\\tests\\unit\\bokeh\\embed\\ext_package_no_main\\dist\\ext_package_no_main.js'
FAILED tests/unit/bokeh/embed/test_bundle.py::Test_bundle_ext_package_no_main::test_with_CDN_resources - FileNotFoundError: [Errno 2] No such file or directory: 'D:\\a\\bokeh\\bokeh\\tests\\unit\\bokeh\\embed\\ext_package_no_main\\dist\\ext_package_no_main.js'
=== 4 failed, 11162 passed, 23 skipped, 129 deselected in 283.34s (0:04:43) ===
```

--- a/.github/workflows/bokeh-ci.yml
+++ b/.github/workflows/bokeh-ci.yml
@@ -106,6 +106,8 @@ jobs:
         run: |
           conda info
           conda list
+          echo "node $(node --version)"
+          echo "npm $(npm --version)"
 
       - name: Run codebase checks
         run: pytest --color=yes tests/codebase
@@ -150,6 +152,8 @@ jobs:
         run: |
           conda info
           conda list
+          echo "node $(node --version)"
+          echo "npm $(npm --version)"
           chromium --version
 
       - name: Start chrome headless
@@ -196,6 +200,8 @@ jobs:
         run: |
           conda info
           conda list
+          echo "node $(node --version)"
+          echo "npm $(npm --version)"
 
       - name: Run tests
         run: pytest -v --cov=bokeh --cov-report=xml --tb=short --driver chrome --color=yes tests/integration
@@ -237,6 +243,8 @@ jobs:
         run: |
           conda info
           conda list
+          echo "node $(node --version)"
+          echo "npm $(npm --version)"
 
       - name: Test defaults
         run: pytest tests/test_defaults.py
@@ -281,6 +289,8 @@ jobs:
         run: |
           conda info
           conda list
+          echo "node $(node --version)"
+          echo "npm $(npm --version)"
 
       - name: Run tests
         run: pytest -m "not sampledata" --cov=bokeh --cov-report=xml --color=yes tests/unit
@@ -310,6 +320,8 @@ jobs:
         run: |
           conda info
           conda list
+          echo "node $(node --version)"
+          echo "npm $(npm --version)"
 
       - name: Build docs
         run: bash scripts/ci/build_docs.sh
@@ -341,6 +353,8 @@ jobs:
         run: |
           conda info
           conda list
+          echo "node $(node --version)"
+          echo "npm $(npm --version)"
 
       - name: Run tests
         run: bash scripts/ci/run_downstream_tests.sh

--- a/.github/workflows/composite/test-setup/action.yml
+++ b/.github/workflows/composite/test-setup/action.yml
@@ -59,6 +59,11 @@ runs:
       working-directory: ./bokehjs/build/dist/
       run: ln -s $(ls -t bokeh-bokehjs-*.tgz | head -n 1) bokeh-bokehjs.tgz
 
+    - name: Upgrade npm
+      shell: bash -l {0}
+      run: |
+        npm install --location=global npm
+
     - name: Cache node modules
       if: ${{ inputs.source-tree == 'keep' }}
       uses: actions/cache@v4

--- a/bokehjs/make/index.js
+++ b/bokehjs/make/index.js
@@ -6,8 +6,9 @@ const {join, dirname, basename} = require("path")
 function npm_install() {
   const npm = process.platform != "win32" ? "npm" : "npm.cmd"
   const {status} = cp.spawnSync(npm, ["install"], {stdio: "inherit"})
-  if (status !== 0)
+  if (status !== 0) {
     process.exit(status)
+  }
 }
 
 if (!fs.existsSync("node_modules/")) {
@@ -35,14 +36,16 @@ if (!semver.satisfies(npm_version, engines.npm)) {
 function is_up_to_date(file) {
   const hash_file = join(dirname(file), `.${basename(file)}`)
 
-  if (!fs.existsSync(hash_file))
+  if (!fs.existsSync(hash_file)) {
     return false
+  }
 
   const old_hash = fs.readFileSync(hash_file)
 
-  const new_hash = crypto.createHash("sha256")
-                         .update(fs.readFileSync(file))
-                         .digest("hex")
+  const new_hash = crypto
+    .createHash("sha256")
+    .update(fs.readFileSync(file))
+    .digest("hex")
 
   return old_hash == new_hash
 }
@@ -58,7 +61,7 @@ for (const workspace of ["", ...workspaces]) {
 
 const {register} = require("ts-node")
 
-process.on('uncaughtException', function(err) {
+process.on("uncaughtException", function(err) {
   console.error(err)
   process.exit(1)
 })
@@ -74,5 +77,6 @@ tsconfig_paths.register({
   },
 })
 
-if (require.main != null)
+if (require.main != null) {
   require("./main")
+}

--- a/bokehjs/make/index.js
+++ b/bokehjs/make/index.js
@@ -4,8 +4,9 @@ const fs = require("fs")
 const {join, dirname, basename} = require("path")
 
 function npm_install() {
-  const npm = process.platform != "win32" ? "npm" : "npm.cmd"
-  const {status} = cp.spawnSync(npm, ["install"], {stdio: "inherit"})
+  const is_windows = process.platform == "win32"
+  const npm = is_windows ? "npm.cmd" : "npm"
+  const {status} = cp.spawnSync(npm, ["install"], {stdio: "inherit", shell: is_windows})
   if (status !== 0) {
     process.exit(status)
   }

--- a/bokehjs/make/main.ts
+++ b/bokehjs/make/main.ts
@@ -1,4 +1,8 @@
 import yargs from "yargs"
+import cp from "child_process"
+
+import chalk from "chalk"
+const {magenta} = chalk
 
 export const argv = yargs.help(false).options({
   // paths
@@ -32,7 +36,14 @@ export const argv = yargs.help(false).options({
 import {task, run, log, task_names, show_error, show_failure} from "./task"
 import "./tasks"
 
+const node_version = process.version
+
+function npm_version(): string {
+  return cp.execSync("npm --version").toString().trim()
+}
+
 async function main(): Promise<void> {
+  log(`Using nodejs ${magenta(node_version)} and npm ${magenta(npm_version())}`)
   const {_} = argv
   if (_.length != 0 && _[0] == "help") {
     log(`tasks: ${task_names().filter((name) => !name.includes(":")).join(", ")}`)

--- a/bokehjs/make/tasks/pack.ts
+++ b/bokehjs/make/tasks/pack.ts
@@ -10,8 +10,9 @@ function npm_pack() {
     fs.mkdirSync(dist, {recursive: true})
   }
 
-  const npm = process.platform != "win32" ? "npm" : "npm.cmd"
-  const {status, stdout, stderr} = cp.spawnSync(npm, ["pack", `--pack-destination=${dist}`], {stdio: "pipe", encoding: "utf-8"})
+  const is_windows = process.platform == "win32"
+  const npm = is_windows ? "npm.cmd" : "npm"
+  const {status, stdout, stderr} = cp.spawnSync(npm, ["pack", `--pack-destination=${dist}`], {stdio: "pipe", encoding: "utf-8", shell: is_windows})
   if (status !== 0) {
     console.error(stdout)
     console.error(stderr)

--- a/bokehjs/make/tasks/pack.ts
+++ b/bokehjs/make/tasks/pack.ts
@@ -18,8 +18,8 @@ function npm_pack() {
     throw new BuildError("pack", `failed to run '${npm} pack'`)
   }
 
-  const tgz = stdout.trim()
-  if (!tgz.endsWith(".tgz")) {
+  const tgz = stdout.trim().split("\n").at(-1)?.trim()
+  if (tgz == null || !tgz.endsWith(".tgz")) {
     throw new BuildError("pack", "can't find tgz archive name in the output")
   }
 

--- a/bokehjs/src/compiler/build.ts
+++ b/bokehjs/src/compiler/build.ts
@@ -61,6 +61,12 @@ function count_files(files: string[]): string {
   return n == 1 ? str : `${str}s`
 }
 
+const node_version = process.version
+
+function npm_version(base_dir: Path): string {
+  return cp.execSync("npm --version", {cwd: base_dir}).toString().trim()
+}
+
 function npm_install(base_dir: Path): void {
   const npm = process.platform != "win32" ? "npm" : "npm.cmd"
   const {status} = cp.spawnSync(npm, ["install"], {stdio: "inherit", cwd: base_dir})
@@ -68,6 +74,11 @@ function npm_install(base_dir: Path): void {
     print(`${cyan("npm install")} failed with exit code ${red(`${status}`)}.`)
     process.exit(status)
   }
+}
+
+function preamble(base_dir: Path): void {
+  print(`Using nodejs ${magenta(node_version)} and npm ${magenta(npm_version(base_dir))}`)
+  print(`Working directory: ${cyan(base_dir)}`)
 }
 
 type Metadata = {
@@ -109,7 +120,7 @@ export type InitOptions = {
 }
 
 export async function init(base_dir: Path, _bokehjs_dir: Path, base_setup: InitOptions): Promise<boolean> {
-  print(`Working directory: ${cyan(base_dir)}`)
+  preamble(base_dir)
 
   const setup: Required<InitOptions> = {
     interactive: base_setup.interactive ?? false,
@@ -218,7 +229,7 @@ export type BuildOptions = {
 }
 
 export async function build(base_dir: Path, bokehjs_dir: Path, base_setup: BuildOptions): Promise<boolean> {
-  print(`Working directory: ${cyan(base_dir)}`)
+  preamble(base_dir)
 
   const setup: Required<BuildOptions> = {
     verbose: base_setup.verbose ?? false,

--- a/bokehjs/src/compiler/build.ts
+++ b/bokehjs/src/compiler/build.ts
@@ -68,8 +68,9 @@ function npm_version(base_dir: Path): string {
 }
 
 function npm_install(base_dir: Path): void {
-  const npm = process.platform != "win32" ? "npm" : "npm.cmd"
-  const {status} = cp.spawnSync(npm, ["install"], {stdio: "inherit", cwd: base_dir})
+  const is_windows = process.platform == "win32"
+  const npm = is_windows ? "npm.cmd" : "npm"
+  const {status} = cp.spawnSync(npm, ["install"], {stdio: "inherit", cwd: base_dir, shell: is_windows})
   if (status != null && status != 0) {
     print(`${cyan("npm install")} failed with exit code ${red(`${status}`)}.`)
     process.exit(status)

--- a/conda/environment-test-downstream.yml
+++ b/conda/environment-test-downstream.yml
@@ -19,3 +19,4 @@ dependencies:
   - pytest >=7.0
   - pytest-timeout
   - scipy
+  - nodejs 20.*

